openhands/runtime/action_execution_server.py

```diff
@@ -1067,8 +1068,69 @@ async def list_files(request: Request):
             return JSONResponse(content=sorted_entries)
 
         except Exception as e:
-            logger.error(f'Error listing files: {e}')
+            logger.error(f'Error listing files: {e}', exc_info=True)
             return JSONResponse(content=[])
 
     logger.debug(f'Starting action execution API on port {args.port}')
-    run(app, host='0.0.0.0', port=args.port)
+    # When LOG_JSON=1, provide a JSON log config to Uvicorn so error/access logs are structured
+    log_config = None
+    if os.getenv('LOG_JSON', '0') in ('1', 'true', 'True'):
```

Comment:
This log config is rather long to be sitting in the middle of the flow here. Probably it should be extracted into a constant or function and referenced from here.